gradio/documentation.py

```diff
@@ -0,0 +1,146 @@
+import inspect
+
+classes_to_document = {}
+gl = {"mode": None}
+
+
+def document_mode(m):
+    gl["mode"] = m
+    if m not in classes_to_document:
+        classes_to_document[m] = []
+
+
+def document(*fns):
+    def inner_doc(cls):
+        classes_to_document[gl["mode"]].append((cls, fns))
+        return cls
+
+    return inner_doc
+
+
+def document_fn(fn):
+    doc_str = inspect.getdoc(fn)
+    doc_lines = doc_str.split("\n")
+    signature = inspect.signature(fn)
+    description, parameters, returns, examples = [], {}, [], []
+    mode = "description"
+    for line in doc_lines:
+        if line == "Parameters:":
```

Comment:
Perhaps make this and following lines code-insensitive.
```suggestion
        if line.lower() == "parameters:":
```

Reply:
Actually I'd rather we be strict on format, and maybe even throw error/warnings on unrecognized formats

Reply:
But when would this error be thrown? If one of us makes a change to the documentation and breaks this format, would we have to build the website to know (that would be pretty inconvenient)?

Reply:
Ok saw your later comment about unit testing. Then it's fine!